controller/src/main/java/io/pravega/controller/store/stream/tables/StreamTruncationRecord.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.stream.tables;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Data class for storing information about stream's truncation point.
+ */
+@AllArgsConstructor(access = AccessLevel.PACKAGE)
+public class StreamTruncationRecord implements Serializable {
+    public static final StreamTruncationRecord EMPTY = new StreamTruncationRecord(Collections.emptyMap(),
+            Collections.emptyMap(), Collections.emptySet(), Collections.emptySet());
+
+    /**
+     * Stream cut that is applied as part of this truncation.
+     */
+    private final Map<Integer, Long> streamCut;
```

Comment:
This and the map below should be immutable maps so that the caller can't modify them after passing them in.

Reply:
done